packages/swirl-components/src/components/swirl-tree-navigation-item/swirl-tree-navigation-item.tsx

```diff
@@ -7,40 +18,179 @@ import classnames from "classnames";
   tag: "swirl-tree-navigation-item",
 })
 export class SwirlTreeNavigationItem {
-  @Element() el: HTMLElement;
+  @Element() el: HTMLSwirlTreeNavigationItemElement;
 
   @Prop() active?: boolean;
   @Prop() href?: string;
   @Prop() icon?: string;
   @Prop() label!: string;
   @Prop() target?: string;
+  @Prop() navigationItemId!: string;
+  @Prop() level?: number = 1;
+  @Prop() expandable?: boolean = true;
+  @Prop() external?: boolean = false;
+
+  @Event() expandedChange!: EventEmitter<boolean>;
+  @State() expanded = false;
+  @State() hasChildren = false;
+
+  private buttonId = `${this.navigationItemId}-button`;
+  private childrenId = `${this.navigationItemId}-children`;
+
+  componentWillLoad() {
+    this.checkForChildren();
+  }
+
+  @Watch("expanded")
+  handleExpandedChange() {
+    this.expandedChange.emit(this.expanded);
+  }
+
+  private checkForChildren() {
+    this.hasChildren = Boolean(
+      this.el.querySelector("swirl-tree-navigation-item")
+    );
+  }
+
+  @Method()
+  async expand() {
+    if (this.expanded || !this.expandable || !this.hasChildren) {
+      return;
+    }
+    this.expanded = true;
+  }
+
+  @Method()
+  async collapse() {
+    if (!this.expanded || !this.expandable || !this.hasChildren) {
+      return;
+    }
+    this.expanded = false;
+  }
+
+  private onKeyDown = (event: KeyboardEvent) => {
+    switch (event.key) {
+      case "Enter":
```

Comment:
Issue: Enter should not toggle the item, but always perform the default action (activating the link). See https://www.w3.org/WAI/ARIA/apg/patterns/treeview/